src/dot-protocol/merlin_dot_protocol.mli

```diff
@@ -74,19 +74,53 @@ end
 
 type directive = Directive.Processed.t
 
-module Commands : sig
-  type t = File of string | Halt | Unknown
-
-  val read_input : in_channel -> t
-  val send_file : out_channel:out_channel -> string -> unit
-end
-
 type read_error =
   | Unexpected_output of string
   | Csexp_parse_error of string
 
-(** [read inc] reads one csexp from the channel [inc] and returns the list of
-  directives it represents *)
-val read : in_channel:in_channel -> (directive list, read_error) Merlin_utils.Std.Result.t
+module Make (IO : sig
```

Comment:
I think we should add a docstring here to explain that this is provided for projects using merlin as a library

Reply:
Sure, I've added a docstring. Do you think it's clear enough?